ios/native/RCTAppDelegate+Reanimated.mm

```diff
@@ -0,0 +1,28 @@
+#if REACT_NATIVE_MINOR_VERSION >= 72 && !defined(RCT_NEW_ARCH_ENABLED) && !defined(DONT_AUTOINSTALL_REANIMATED)
+
+#import <RNReanimated/RCTAppDelegate+Reanimated.h>
+#import <RNReanimated/REAInitializer.h>
+#import <objc/runtime.h>
+
+@implementation RCTAppDelegate (Reanimated)
+
+- (std::unique_ptr<facebook::react::JSExecutorFactory>)swizzled_jsExecutorFactoryForBridge:(RCTBridge *)bridge
+{
+  reanimated::REAInitializer(bridge);
+  return [self swizzled_jsExecutorFactoryForBridge:bridge]; // call the original method
+}
+
++ (void)load
+{
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    Class cls = [self class];
+    Method originalMethod = class_getInstanceMethod(cls, @selector(jsExecutorFactoryForBridge:));
+    Method swizzledMethod = class_getInstanceMethod(cls, @selector(swizzled_jsExecutorFactoryForBridge:));
+    method_exchangeImplementations(originalMethod, swizzledMethod);
```

Comment:
Will this still work if the instance doesn't implement `jsExecutorFactoryForBridge:`?

I wonder if you should use something like this: https://github.com/microsoft/react-native-test-app/blob/2e5d0e64535230ed07a4ecf398387057b34b7121/ios/ReactTestApp/React%2BCompatibility.m#L7

I think you should also use a different prefix for the method name to avoid potential name clashes, e.g. `reanimated_jsExecutorFactoryForBridge`.

Reply:
Hey @tido64, thanks for the review! I've renamed the suffix in d018b2cc511df8165157e306a791d773bce2901d as you suggested 😄 

As for the case when `jsExecutorFactoryForBridge:` is not implemented, indeed we don't handle it properly now but it looks like since 0.72.0-rc.4 this is always implemented in `RCTAppDelegate`, at least in greenfield apps. We could add a conditional on `didAddMethod` as you proposed but I'm not sure what would happen in the line where we call `[self reanimated_jsExecutorFactoryForBridge:bridge]` then. I need some more time to learn and think about it.

Thanks for the code pointer to `RTASwizzleSelector`, would you mind if we copy it to Reanimated codebase as a utility function? We also need to swizzle some methods of ViewControllers for Shared Element Transitions. 

Finally, I'm curious to know what's the reason to use `+initialize` [here](https://github.com/microsoft/react-native-test-app/blob/2e5d0e64535230ed07a4ecf398387057b34b7121/ios/ReactTestApp/React%2BCompatibility.m#L38), I found one article that recommends to perform swizzling always in `+load` method ([link](https://nshipster.com/method-swizzling/#load-vs-initialize)).

Reply:
> Thanks for the code pointer to `RTASwizzleSelector`, would you mind if we copy it to Reanimated codebase as a utility function? We also need to swizzle some methods of ViewControllers for Shared Element Transitions.

Feel free to copy it 👍

> Finally, I'm curious to know what's the reason to use `+initialize` [here](https://github.com/microsoft/react-native-test-app/blob/2e5d0e64535230ed07a4ecf398387057b34b7121/ios/ReactTestApp/React%2BCompatibility.m#L38), I found one article that recommends to perform swizzling always in `+load` method ([link](https://nshipster.com/method-swizzling/#load-vs-initialize)).

The reason behind using `+initialize` is because we only want to swizzle if the class is actually used. In our case, race condition isn't really a concern since React doesn't initialize modules on multiple threads (at least not currently). We also use `dispatch_once` here so its atomicity is ensured.

Reply:
> Feel free to copy it 👍

Thanks!

> The reason behind using `+initialize` is because we only want to swizzle if the class is actually used. In our case, race condition isn't really a concern since React doesn't initialize modules on multiple threads (at least not currently). We also use `dispatch_once` here so its atomicity is ensured.

That makes perfect sense, thanks!

Reply:
> Will this still work if the instance doesn't implement `jsExecutorFactoryForBridge:`?

@tido64 I'm still thinking about this question. Our aim is to automatically initialize Reanimated for greenfield apps; if some custom app doesn't use `RCTAppDelegate`, it needs to perform initialization on its own. It looks like since 0.72.0-rc.4 `jsExecutorFactoryForBridge:` is already implemented in RCTAppDelegate ([link](https://github.com/facebook/react-native/blob/e11396e9980761a205d5cd96d5129f5385fd074b/packages/react-native/Libraries/AppDelegate/RCTAppDelegate.mm#L143-L159)). This means that we can assume that `didAddMethod` returns `false` which reduces to the code that we currently have in this PR, or am I missing something?

Reply:
> As far as I can tell, Reanimated only needs to swizzle `jsExecutorFactoryForBridge:` on Paper. Is that correct? 

Yes, that's correct. This PR introduces swizzling in `RCTAppDelegate` only for React Native 0.72+:

https://github.com/software-mansion/react-native-reanimated/blob/d018b2cc511df8165157e306a791d773bce2901d/ios/native/RCTAppDelegate%2BReanimated.mm#L1

For 0.71 and older we will still use the old way which is to implement `jsExecutorFactoryForBridge:` method for `UIResponder`:

https://github.com/software-mansion/react-native-reanimated/blob/d018b2cc511df8165157e306a791d773bce2901d/ios/native/UIResponder%2BReanimated.mm#L1

Reply:
Ah, I missed this part. Then it looks like you should be good for now.